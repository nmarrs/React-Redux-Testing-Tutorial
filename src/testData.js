export let testTask = {
  id: 1,
  title: 'Unload merchandise',
  priority: 3,
  timeEstimate: 2,
  notes: 'Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 150
}

export let testTask1 = {
  id: 2,
  title: 'Stock shelves with unpacked items',
  priority: 1,
  timeEstimate: 10,
  notes: 'Take merchandise items and stock shelves with them. Be sure to place items in correct areas.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 10
}

export let testTask2 = {
  id: 3,
  title: 'Stock shelves with unpacked items',
  priority: 5,
  timeEstimate: 2,
  notes: 'Take merchandise items and stock shelves with them. Be sure to place items in correct areas.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 250
}
