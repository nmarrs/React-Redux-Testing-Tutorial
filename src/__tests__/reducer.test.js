import { rootReducer } from ".././index";
import { testTask, testTask2 } from "../testData";

describe("reducer", () => {
  describe("INITIAL_STATE", () => {
    it("is correct", () => {
      const action = { type: "dummy_action" };

      expect(rootReducer(undefined, action)).toMatchSnapshot();
    });
  });

  describe("TOGGLE_CREATE_TASK_MODAL", () => {
    it("is correct", () => {
      let showCreateTaskModal = true;
      const action = {
        type: "TOGGLE_CREATE_TASK_MODAL",
        payload: showCreateTaskModal,
        task: testTask
      };

      expect(rootReducer(undefined, action)).toMatchSnapshot();
    });
  });

  // INSERT TEST FOR `TOGGLE_EDIT_TASK_MODAL` HERE!!

  describe("TOGGLE_IS_ADMIN_VIEW", () => {
    it("is correct", () => {
      const action = { type: "TOGGLE_IS_ADMIN_VIEW" };

      expect(rootReducer(undefined, action)).toMatchSnapshot();
    });
  });

  describe("ASSIGN_TASKS_TO_USER", () => {
    it("is correct", () => {
      const action = { type: "ASSIGN_TASKS_TO_USER", payload: [testTask] };

      expect(rootReducer(undefined, action)).toMatchSnapshot();
    });
  });

  describe("CREATE_TEMPLATE_TASK", () => {
    it("is correct", () => {
      const action = { type: "CREATE_TEMPLATE_TASK", payload: testTask2 };

      let returnedState = rootReducer(undefined, action);
      expect(returnedState).toMatchSnapshot();
      // Check to ensure sorting works as expected
      expect(returnedState.templateTasks[0].rankWeight).toBe(250);
    });
  });

  describe("CREATE_TASK", () => {
    it("is correct", () => {
      const action = { type: "CREATE_TASK", payload: testTask2 };

      expect(rootReducer(undefined, action)).toMatchSnapshot();
    });

    it("sorts correctly", () => {
      const action1 = { type: "ASSIGN_TASKS_TO_USER", payload: [testTask] };
      const action2 = { type: "CREATE_TASK", payload: testTask2 };

      // First add initial template tasks to user
      let firstState = rootReducer(undefined, action1);

      let nextState = rootReducer(firstState, action2);
      expect(nextState).toMatchSnapshot();
      // Check to ensure sorting works as expected
      expect(nextState.user.tasks[0].rankWeight).toBe(250);
    });
  });

  describe("EDIT_TEMPLATE_TASK", () => {
    it("is correct", () => {
      let changedTask = { ...testTask };
      changedTask.feedback = "Some example feedback.";
      const action = { type: "EDIT_TEMPLATE_TASK", payload: changedTask };

      let returnedState = rootReducer(undefined, action);
      expect(returnedState).toMatchSnapshot();
    });
  });

  describe("EDIT_TASK", () => {
    it("is correct", () => {
      const action1 = { type: "ASSIGN_TASKS_TO_USER", payload: [testTask] };
      let changedTask = { ...testTask };
      changedTask.feedback = "Some example feedback.";
      const action2 = { type: "EDIT_TASK", payload: changedTask };

      // First add initial template tasks to user
      let firstState = rootReducer(undefined, action1);

      let nextState = rootReducer(firstState, action2);
      expect(nextState).toMatchSnapshot();
    });
  });

  describe("SORT_TASKS_BY_RANK", () => {
    it("is correct", () => {
      let templateTasks = [testTask];
      let userTasks = [testTask2];
      const action = {
        type: "SORT_TASKS_BY_RANK",
        sortedTemplateTasks: templateTasks,
        sortedUserTasks: userTasks
      };

      let returnedState = rootReducer(undefined, action);
      expect(returnedState).toMatchSnapshot();
    });
  });
});
