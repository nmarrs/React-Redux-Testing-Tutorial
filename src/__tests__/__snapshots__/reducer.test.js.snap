// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer ASSIGN_TASKS_TO_USER is correct 1`] = `
Object {
  "createTaskModalOpen": false,
  "currentTask": Object {},
  "editTaskModalOpen": false,
  "isAdminView": true,
  "templateTasks": Array [
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 1,
      "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
      "priority": 3,
      "rankWeight": 150,
      "startTime": "",
      "timeEstimate": 2,
      "title": "Unload merchandise",
    },
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 2,
      "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
      "priority": 1,
      "rankWeight": 10,
      "startTime": "",
      "timeEstimate": 10,
      "title": "Stock shelves with unpacked items",
    },
  ],
  "user": Object {
    "tasks": Array [
      Object {
        "completionTime": "",
        "currentStatus": "To Do",
        "endTime": "",
        "feedback": "",
        "id": 1,
        "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
        "priority": 3,
        "rankWeight": 150,
        "startTime": "",
        "timeEstimate": 2,
        "title": "Unload merchandise",
      },
    ],
  },
}
`;

exports[`reducer CREATE_TASK is correct 1`] = `
Object {
  "createTaskModalOpen": false,
  "currentTask": Object {},
  "editTaskModalOpen": false,
  "isAdminView": true,
  "templateTasks": Array [
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 1,
      "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
      "priority": 3,
      "rankWeight": 150,
      "startTime": "",
      "timeEstimate": 2,
      "title": "Unload merchandise",
    },
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 2,
      "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
      "priority": 1,
      "rankWeight": 10,
      "startTime": "",
      "timeEstimate": 10,
      "title": "Stock shelves with unpacked items",
    },
  ],
  "user": Object {
    "tasks": Array [
      Object {
        "completionTime": "",
        "currentStatus": "To Do",
        "endTime": "",
        "feedback": "",
        "id": 3,
        "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
        "priority": 5,
        "rankWeight": 250,
        "startTime": "",
        "timeEstimate": 2,
        "title": "Stock shelves with unpacked items",
      },
    ],
  },
}
`;

exports[`reducer CREATE_TASK sorts correctly 1`] = `
Object {
  "createTaskModalOpen": false,
  "currentTask": Object {},
  "editTaskModalOpen": false,
  "isAdminView": true,
  "templateTasks": Array [
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 1,
      "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
      "priority": 3,
      "rankWeight": 150,
      "startTime": "",
      "timeEstimate": 2,
      "title": "Unload merchandise",
    },
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 2,
      "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
      "priority": 1,
      "rankWeight": 10,
      "startTime": "",
      "timeEstimate": 10,
      "title": "Stock shelves with unpacked items",
    },
  ],
  "user": Object {
    "tasks": Array [
      Object {
        "completionTime": "",
        "currentStatus": "To Do",
        "endTime": "",
        "feedback": "",
        "id": 3,
        "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
        "priority": 5,
        "rankWeight": 250,
        "startTime": "",
        "timeEstimate": 2,
        "title": "Stock shelves with unpacked items",
      },
      Object {
        "completionTime": "",
        "currentStatus": "To Do",
        "endTime": "",
        "feedback": "",
        "id": 1,
        "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
        "priority": 3,
        "rankWeight": 150,
        "startTime": "",
        "timeEstimate": 2,
        "title": "Unload merchandise",
      },
    ],
  },
}
`;

exports[`reducer CREATE_TEMPLATE_TASK is correct 1`] = `
Object {
  "createTaskModalOpen": false,
  "currentTask": Object {},
  "editTaskModalOpen": false,
  "isAdminView": true,
  "templateTasks": Array [
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 3,
      "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
      "priority": 5,
      "rankWeight": 250,
      "startTime": "",
      "timeEstimate": 2,
      "title": "Stock shelves with unpacked items",
    },
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 1,
      "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
      "priority": 3,
      "rankWeight": 150,
      "startTime": "",
      "timeEstimate": 2,
      "title": "Unload merchandise",
    },
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 2,
      "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
      "priority": 1,
      "rankWeight": 10,
      "startTime": "",
      "timeEstimate": 10,
      "title": "Stock shelves with unpacked items",
    },
  ],
  "user": Object {
    "tasks": Array [],
  },
}
`;

exports[`reducer EDIT_TASK is correct 1`] = `
Object {
  "createTaskModalOpen": false,
  "currentTask": Object {},
  "editTaskModalOpen": false,
  "isAdminView": true,
  "templateTasks": Array [
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 1,
      "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
      "priority": 3,
      "rankWeight": 150,
      "startTime": "",
      "timeEstimate": 2,
      "title": "Unload merchandise",
    },
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 2,
      "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
      "priority": 1,
      "rankWeight": 10,
      "startTime": "",
      "timeEstimate": 10,
      "title": "Stock shelves with unpacked items",
    },
  ],
  "user": Object {
    "tasks": Array [
      Object {
        "completionTime": "",
        "currentStatus": "To Do",
        "endTime": "",
        "feedback": "Some example feedback.",
        "id": 1,
        "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
        "priority": 3,
        "rankWeight": 150,
        "startTime": "",
        "timeEstimate": 2,
        "title": "Unload merchandise",
      },
    ],
  },
}
`;

exports[`reducer EDIT_TEMPLATE_TASK is correct 1`] = `
Object {
  "createTaskModalOpen": false,
  "currentTask": Object {},
  "editTaskModalOpen": false,
  "isAdminView": true,
  "templateTasks": Array [
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "Some example feedback.",
      "id": 1,
      "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
      "priority": 3,
      "rankWeight": 150,
      "startTime": "",
      "timeEstimate": 2,
      "title": "Unload merchandise",
    },
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 2,
      "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
      "priority": 1,
      "rankWeight": 10,
      "startTime": "",
      "timeEstimate": 10,
      "title": "Stock shelves with unpacked items",
    },
  ],
  "user": Object {
    "tasks": Array [],
  },
}
`;

exports[`reducer INITIAL_STATE is correct 1`] = `
Object {
  "createTaskModalOpen": false,
  "currentTask": Object {},
  "editTaskModalOpen": false,
  "isAdminView": true,
  "templateTasks": Array [
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 1,
      "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
      "priority": 3,
      "rankWeight": 150,
      "startTime": "",
      "timeEstimate": 2,
      "title": "Unload merchandise",
    },
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 2,
      "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
      "priority": 1,
      "rankWeight": 10,
      "startTime": "",
      "timeEstimate": 10,
      "title": "Stock shelves with unpacked items",
    },
  ],
  "user": Object {
    "tasks": Array [],
  },
}
`;

exports[`reducer SORT_TASKS_BY_RANK is correct 1`] = `
Object {
  "createTaskModalOpen": false,
  "currentTask": Object {},
  "editTaskModalOpen": false,
  "isAdminView": true,
  "templateTasks": Array [
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 1,
      "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
      "priority": 3,
      "rankWeight": 150,
      "startTime": "",
      "timeEstimate": 2,
      "title": "Unload merchandise",
    },
  ],
  "user": Object {
    "tasks": Array [
      Object {
        "completionTime": "",
        "currentStatus": "To Do",
        "endTime": "",
        "feedback": "",
        "id": 3,
        "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
        "priority": 5,
        "rankWeight": 250,
        "startTime": "",
        "timeEstimate": 2,
        "title": "Stock shelves with unpacked items",
      },
    ],
  },
}
`;

exports[`reducer TOGGLE_CREATE_TASK_MODAL is correct 1`] = `
Object {
  "createTaskModalOpen": true,
  "currentTask": Object {},
  "editTaskModalOpen": false,
  "isAdminView": true,
  "templateTasks": Array [
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 1,
      "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
      "priority": 3,
      "rankWeight": 150,
      "startTime": "",
      "timeEstimate": 2,
      "title": "Unload merchandise",
    },
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 2,
      "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
      "priority": 1,
      "rankWeight": 10,
      "startTime": "",
      "timeEstimate": 10,
      "title": "Stock shelves with unpacked items",
    },
  ],
  "user": Object {
    "tasks": Array [],
  },
}
`;

exports[`reducer TOGGLE_EDIT_TASK_MODAL is correct 1`] = `
Object {
  "createTaskModalOpen": false,
  "currentTask": undefined,
  "editTaskModalOpen": true,
  "isAdminView": true,
  "templateTasks": Array [
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 1,
      "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
      "priority": 3,
      "rankWeight": 150,
      "startTime": "",
      "timeEstimate": 2,
      "title": "Unload merchandise",
    },
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 2,
      "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
      "priority": 1,
      "rankWeight": 10,
      "startTime": "",
      "timeEstimate": 10,
      "title": "Stock shelves with unpacked items",
    },
  ],
  "user": Object {
    "tasks": Array [],
  },
}
`;

exports[`reducer TOGGLE_IS_ADMIN_VIEW is correct 1`] = `
Object {
  "createTaskModalOpen": false,
  "currentTask": Object {},
  "editTaskModalOpen": false,
  "isAdminView": false,
  "templateTasks": Array [
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 1,
      "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
      "priority": 3,
      "rankWeight": 150,
      "startTime": "",
      "timeEstimate": 2,
      "title": "Unload merchandise",
    },
    Object {
      "completionTime": "",
      "currentStatus": "To Do",
      "endTime": "",
      "feedback": "",
      "id": 2,
      "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
      "priority": 1,
      "rankWeight": 10,
      "startTime": "",
      "timeEstimate": 10,
      "title": "Stock shelves with unpacked items",
    },
  ],
  "user": Object {
    "tasks": Array [],
  },
}
`;
